File: InterviewCampProblems/InterviewCampProblems.Test/UnitTest8.cs
/*
You are given an array of integers. Rearrange the array so that all zeroes are at the beginning of the array.For example,a = [4,2,0,1,0,3,0] -> [0, 0, 0, 1, 2, 3, 4]
*/

using System;
using InterviewCampProblems.Code;
using Xunit;

namespace InterviewCampProblems.Test {
    public class UnitTest8 {
        [Fact]
        public void Test1 () {

            int[] expected = new int[] { 0, 0, 0, 1, 2, 3, 4 };
            int[] arr = new int[] { 4, 2, 0, 1, 0, 3, 0 };
            var actual = Class8.MoveZerosToStart (arr);
            Assert.Equal (expected, actual);
        }

        [Fact]
        public void Test2 () {

            int[] expected = new int[] { 4 };
            int[] arr = new int[] { 4 };
            var actual = Class8.MoveZerosToStart (arr);
            Assert.Equal (expected, actual);
        }

        [Fact]
        public void Test3 () {

            int[] expected = new int[] { 0 };
            int[] arr = new int[] { 0 };
            var actual = Class8.MoveZerosToStart (arr);
            Assert.Equal (expected, actual);
        }

        [Fact]
        public void Test4 () {

            int[] expected = null;
            int[] arr = null;
            var actual = Class8.MoveZerosToStart (arr);
            Assert.Equal (expected, actual);
        }
    }
}

File: Java/InterviewCamp/src/test/java/UnitTest8.java
//You are given an array of integers. Rearrange the array so that all zeroes are at the beginning of the array.For example,a = [4,2,0,1,0,3,0] -> [0,0,0,1,2,3,4]

import org.junit.Assert;
import org.junit.Test;

public class UnitTest8 {

    @Test
    public void When_Input_Correct() {

        //Arrange
        int[] arr = new int[] {4,2,0,1,0,3,0};
        int[] outputArr = new int[] {0,0,0,1,2,3,4};

        //Act
        int[] output = Class8.RearrangeArray(arr);

        //Assert
        Assert.assertArrayEquals(output,outputArr);

    }
}